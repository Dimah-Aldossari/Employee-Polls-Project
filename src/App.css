* {
  margin: 0px;
  padding: 0px;
}
nav {
  border: 1px solid #ccc;
  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.1);
  padding: 15px;
  background-color: #ccbcdc;
}
ul {
  display: flex;
  justify-content: space-around;
  align-items: center;
  list-style-type: none;
  margin: 0;
  padding: 0;
}
.btn-box {
  width: 80%;
  height: 70%;
  display: flex;
  justify-content: center;
  margin-top: 10px;
}
.btn {
  margin-right: 10px;
  padding: 8px 16px;
  border: none;
  cursor: pointer;
}
.input {
  padding: 8px 16px;
  width: 70%;
  margin: 3px;
}

/* Landing */

.landing {
  background-image: url("https://images.pexels.com/photos/289737/pexels-photo-289737.jpeg?auto=compress&cs=tinysrgb&w=800");
  width: 100%;
  height: 100vh;
  background-repeat: no-repeat;
  background-size: contain;
  background-size: cover;
  margin: 0px;
  text-align: center;
  display: flex;
  align-items: center;
  justify-content: center;
  color: #ccc;
  font-size: larger;
}
.question-card {
  border: 1px solid #ddd;
  padding: 10px;
  margin-bottom: 10px;
  border-radius: 5px;
}
.avatar {
  width: 50px; /* adjust as needed */
  height: 50px; /* adjust as needed */
  border-radius: 50%;
}

.question-card {
  border: 1px solid #ddd;
  padding: 10px;
  margin-bottom: 10px;
  border-radius: 5px;
}

.avatar {
  width: 70px;
  height: 70px;
  margin: 10px 0px;
  border-radius: 50%;
}

.option {
  padding: 10px;
  margin: 5px 0;
  border: 1px solid #ddd;
  cursor: pointer;
}

.option.voted {
  background-color: lightgreen;
}

.vote-thanks {
  color: green;
}

.navbar-list {
  list-style-type: none;
  margin: 0;
  padding: 0;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.navbar-item {
  margin-right: 10px;
}

/* Styling for the Welcome message */
.username {
  font-weight: bold;
  color: #8b61a4; /* Set the username color to a contrasting color */
}

/* Styling for the buttons */
.nav-button {
  background-color: #ff6600; /* Set button background color */
  color: #fff; /* Set button text color to white */
  border: none;
  padding: 10px 20px;
  font-size: 16px;
  cursor: pointer;
  transition: background-color 0.3s;
}

/* Button hover effect */
.nav-button:hover {
  background-color: #ff9933; /* Lighter color on hover */
}

/* Specific styles for each button */
.home {
  background-color: #009933; /* Green for Home button */
}

.add-pool {
  background-color: #0066cc; /* Blue for Add New Pool button */
}

.logout {
  background-color: #cc3300; /* Red for Logout button */
}
/* Define a class for the LeaderBoard button */
.leaderboard-button {
  background-color: #663399; /* Customize the background color */
  color: #fff; /* Set text color to white */
  border: none;
  padding: 10px 20px;
  font-size: 16px;
  cursor: pointer;
  transition: background-color 0.3s;
}

/* Button hover effect for the LeaderBoard button */
.leaderboard-button:hover {
  background-color: #9966cc; /* Lighter color on hover */
}

/* Leaderboard Styles */
.leaderboard {
  margin: 100px auto; /* Center the leaderboard horizontally */
  max-width: 800px; /* Optional: Limit the maximum width of the leaderboard */
}

.leaderboard h2 {
  font-size: 24px;
  margin-bottom: 10px;
}

.leaderboard-list {
  list-style-type: none;
  padding: 0;
  border-collapse: collapse; /* Collapse table borders */
  width: 100%;
}

.leaderboard-item {
  display: table-row;
  border-bottom: 1px solid #ddd; /* Add a bottom border to separate rows */
}

.user-info {
  display: table-cell;
  padding: 10px;
  vertical-align: middle;
}

.user-avatar {
  width: 40px;
  height: 40px;
  border-radius: 50%;
  margin-right: 10px;
}

.user-name {
  font-weight: bold;
}

.user-stats {
  display: table-cell;
  padding: 10px;
  text-align: center;
  vertical-align: middle;
}

/* Style the header row */
.leaderboard-item:first-child {
  background-color: #f2f2f2;
  font-weight: bold;
}

/* Style alternating rows */
.leaderboard-item:nth-child(even) {
  background-color: #f2f2f2;
}

/* Style cells */
.leaderboard-item td {
  padding: 10px;
  border-right: 1px solid #ddd; /* Add right border to cells */
}

/* Remove right border from the last cell in each row */
.leaderboard-item td:last-child {
  border-right: none;
}
.Create-Question {
  text-align: center;
  margin: 10px;
}

.Option {
  display: grid;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  margin-bottom: 10px;
}
