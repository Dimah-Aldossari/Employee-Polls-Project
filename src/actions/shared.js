// actions/shared.js

import { setUsers } from "./usersSlice";
import { setQuestions } from "./questionsSlice";
import { getInitialData } from "../api";

export function handleInitialData() {
    return (dispatch) => {
        return getInitialData().then(({ users, questions }) => {
            dispatch(setUsers(users));        // Corrected usage
            dispatch(setQuestions(questions)); // Corrected usage
        });
    };
}
