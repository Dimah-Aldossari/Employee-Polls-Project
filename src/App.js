
import './App.css'
import { Route, Routes } from 'react-router-dom';
import Navbar from './components/Navbar';
import Home from './components/Home';
import Landing from './components/Landing'
import Login from './components/Login'
import AddQuestion from './components/AddQuestion';
import Error404 from './components/Error404';
import PrivateRoute from './components/PrivateRoute';
import Leaderboard from './components/Leaderboard.js';


function App() {


  return (
    <>
      <Navbar />

      <Routes>

        <Route path="/" exact element={<Landing />} />
        <Route path="/LeaderBoard" element={<PrivateRoute><Leaderboard /></PrivateRoute>} />
        <Route path="/Home" element={<PrivateRoute><Home /></PrivateRoute>} />
        <Route path="/addQuestion" element={<PrivateRoute><AddQuestion /></PrivateRoute>} />
        <Route path="/Login" element={<Login />} />
        <Route path="/*" element={<Error404 />} />

      </Routes>

    </>
  )
}

export default App
