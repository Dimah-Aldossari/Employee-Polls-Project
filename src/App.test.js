import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';
import { Provider } from 'react-redux'
import store from "./Store.js";
import { BrowserRouter } from "react-router-dom";

import { login } from "./actions/Creat_Users";

describe("App", () => {
  test("should render the component", () => {
    const view = render(
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>
    );
    expect(view).toBeDefined();
    // expect(view).toMatchSnapshot();
  });

  test("should show Login page when not logged in", () => {
    const view = render(
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>
    );
    const heading = screen.getByTestId("login-heading");
    expect(heading).toBeInTheDocument();
    expect(view).toMatchSnapshot();

  });
  test("should show Dashboard page when logged in", () => {
    store.dispatch(login({ id: "", password: "" }));

    const view = render(
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>
    );

    const heading = screen.getByTestId("heading");
    expect(heading).toBeInTheDocument();
    expect(view).toMatchSnapshot();

  });

});
