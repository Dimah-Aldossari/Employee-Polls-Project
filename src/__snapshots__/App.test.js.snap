// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should show Dashboard page when logged in 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav>
        <ul>
          <div
            class="welcome"
            data-testid="user-information"
          >
            <span
              class="username"
            >
              welcome: 
            </span>
          </div>
          <div
            class="navbar"
          >
            <div
              class="nav-buttons"
            >
              <button
                class="nav-button home"
              >
                Home
              </button>
              <button
                class="nav-button leaderboard-button"
                data-testid="heading"
              >
                LeaderBoard
              </button>
              <button
                class="nav-button add-pool"
              >
                Add New Pool
              </button>
              <button
                class="nav-button logout"
              >
                Logout
              </button>
            </div>
          </div>
        </ul>
      </nav>
      <div
        class="landing"
      >
        <h1>
          REDUX - PROJECT
        </h1>
      </div>
    </div>
  </body>,
  "container": <div>
    <nav>
      <ul>
        <div
          class="welcome"
          data-testid="user-information"
        >
          <span
            class="username"
          >
            welcome: 
          </span>
        </div>
        <div
          class="navbar"
        >
          <div
            class="nav-buttons"
          >
            <button
              class="nav-button home"
            >
              Home
            </button>
            <button
              class="nav-button leaderboard-button"
              data-testid="heading"
            >
              LeaderBoard
            </button>
            <button
              class="nav-button add-pool"
            >
              Add New Pool
            </button>
            <button
              class="nav-button logout"
            >
              Logout
            </button>
          </div>
        </div>
      </ul>
    </nav>
    <div
      class="landing"
    >
      <h1>
        REDUX - PROJECT
      </h1>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App should show Login page when not logged in 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav>
        <ul>
          <li>
            <input
              class="input"
              placeholder="Username"
              type="text"
              value=""
            />
            <input
              class="input"
              placeholder="Password"
              type="password"
              value=""
            />
            <br />
            <div
              class="btn-box"
            >
              <button
                class="btn"
                data-testid="login-heading"
              >
                Login
              </button>
            </div>
          </li>
        </ul>
      </nav>
      <div
        class="landing"
      >
        <h1>
          REDUX - PROJECT
        </h1>
      </div>
    </div>
  </body>,
  "container": <div>
    <nav>
      <ul>
        <li>
          <input
            class="input"
            placeholder="Username"
            type="text"
            value=""
          />
          <input
            class="input"
            placeholder="Password"
            type="password"
            value=""
          />
          <br />
          <div
            class="btn-box"
          >
            <button
              class="btn"
              data-testid="login-heading"
            >
              Login
            </button>
          </div>
        </li>
      </ul>
    </nav>
    <div
      class="landing"
    >
      <h1>
        REDUX - PROJECT
      </h1>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
